--- mmc.c
+++ mmc.c
@@ -1,8 +1,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#define matSize 1100
-#define maxTest 10
 
 /* Row-major order (Same as Fortran). */
 #define ind(i,j) (j)*matSize+i
@@ -19,7 +17,7 @@ void dummy(double *a, double *b, double *c);
 int main(int argc, char *argv[])
 {
     double sum, tStart, tEnd, tLoop, rate, t;
-    int    i, j, k, tests;
+    int    i, j, k, tests, repeats;
 
     /* Initialize the matrics */
     /* Note that this is *not* in the best order with respect to cache;
@@ -34,6 +32,7 @@ int main(int argc, char *argv[])
     tLoop = 1.0e10;
     for (tests=0; tests<maxTest; tests++) {
 	tStart = mysecond();
+	for (repeats=0; repeats<numRepeats; repeats++)
 	for (i=0; i<matSize; i++)
 	    for (j=0; j<matSize; j++) {
 		sum = 0.0;
@@ -42,7 +41,7 @@ int main(int argc, char *argv[])
 		matC[ind(i,j)] = sum;
 	    }
 	tEnd = mysecond();
-	t = tEnd - tStart;
+	t = (tEnd - tStart) / numRepeats;
 	dummy(matA, matB, matC);
 	if (t < tLoop) tLoop = t;
 	if (matC[ind(0,0)] < 0) {
